Mathlib/Logic/Godel/GodelBetaFunction.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2023 Shogo Saito. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shogo Saito. Adapted for mathlib by Hunter Monroe
+-/
+
+import Mathlib.Data.Nat.ModEq
+import Mathlib.Data.Nat.Prime
+import Mathlib.Data.Nat.Pairing
+import Mathlib.Data.List.FinRange
+import Mathlib.Data.List.MinMax
+
+/-!
+# Gödel's Beta Function Lemma
+
+This file proves Gödel's Beta Function Lemma, used to prove the First Incompleteness Theorem. It
+ permits  quantification over finite sequences of natural numbers in formal theories of arithmetic.
+ This Beta Function has no connection with the unrelated Beta Function defined in analysis.
+
+## Main result
+
+- `beta_function_lemma`: Gödel's Beta Function Lemma.
+
+## Implementation note
+
+This code is a step towards eventually including a proof of Gödel's First Incompleteness Theorem
+and other key results from the repository https://github.com/iehality/lean4-logic.
+
+## References
+
+* [R. Kaye, *Models of Peano arithmetic*][kaye1991]
+* <https://en.wikipedia.org/wiki/G%C3%B6del%27s_%CE%B2_function>
+
+## Tags
+
+Gödel, beta function
+-/
+
+namespace Nat
```

Comment:
These first few nat lemmas that aren't specific to the GodelBetaFunction should be moved to earlier files in the library.

Reply:
I have moved them to Mathlib.Data.Zmod.Coprime